### PR TITLE
Feature/Console Command To Install Nosto Account

### PR DESCRIPTION
**Pull request by supercid**
Using tokens and the existing merchant-id from nosto admin, now the user is able to connect/reconnect their nosto account using a console command.

### CODE REVIEW

File: Console/Command/NostoConfigConnectCommand.php [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    private $nostoHelperScope;
+
+    /**
+     * @var bool
+     */
+    private $isInteractive = true;
+
+    /**
+     * NostoConfigConnectCommand constructor.
+     * @param NostoAccountHelper $accountHelper
```
Comment (supercid) [inline Console/Command/NostoConfigConnectCommand.php]: Missing helperScope DocBlock

File: Console/Command/NostoConfigConnectCommand.php [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+        $tokens = $this->getTokensFromInput($input, $io);
+        if ($this->updateNostoTokens($tokens, $accountId, $io, $scopeCode)) {
+            $io->success('Tokens Sucessfully Configured');
+        } else {
+            $io->error('Could not complete operation');
+        }
+    }
+
+    /**
```
Comment (supercid) [inline Console/Command/NostoConfigConnectCommand.php]: Missing function description

File: Console/Command/NostoConfigConnectCommand.php [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+     * if not, will ask for the remaining parameters.
+     *
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return array of Token objects
+     * @throws \Nosto\NostoException
+     */
+    private function getTokensFromInput(InputInterface $input, SymfonyStyle $io)
+    {
+        $tokens = array();
```
Comment (supercid) [inline Console/Command/NostoConfigConnectCommand.php]: Missing check for non-interactive flag

**Comment by supercid** [comment]
@mridang Command run example:
```
bin/magento nosto:config:connect \
--account-id=magento-e93208e7 \
--sso_token=UoMLRCCcagS22IH0UWGM4TYcPwVU5U2bHO7kpVPvdX5aJV884X1PtyookV7eN4vg \
--products_token=Hc2Uv8MGf8oR9nqwfdLBNEgA1Hq8GxKIY7IsM1WF4rgObPSb7xIQYNGg6hSXM2Iq \
--settings_token=u6NAsH4JTiyIMIH9iP6KMh8U7IiW016U6XiOOzi5vusMNjpr1JheYmK13fGXOJpM \
--rates_token=EZnEmYwtEyY7LEo3BPRJ5w3Dm7Y1SHefqa1AcWt0LrGrSx1JF9oTS7VSPtcprdgt \
--scope-code=default --no-email
```

File: Console/Command/NostoConfigConnectCommand.php [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+        $this->nostoHelperScope = $nostoHelperScope;
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this->setName('nosto:config:connect')
```
Comment (hannunosto) [inline Console/Command/NostoConfigConnectCommand.php]: How about renaming `nosto:account:connect`. I'd "reserve" the config for actual configurations

Comment (supercid) [reply Console/Command/NostoConfigConnectCommand.php]: Good point. Renamed.

File: Console/Command/NostoAccountConnectCommand.php [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+                    'Local Nosto account found for this store view. Override tokens?',
+                    false
+                ):
+               true;
+            if ($confirmOverride) {
+                $account->setTokens($tokens);
+                return $this->accountHelper->saveAccount($account, $store);
+            }
+        } else {
+            $io->note('Local account not found, creating a new one...');
```
Comment (hannunosto) [inline Console/Command/NostoAccountConnectCommand.php]: This might be a bit misleading note. We are not really creating new account are we?

**Pull request merged**